install.sh

```diff
@@ -12,14 +12,44 @@ fi
 
 echo "" 
 echo "Copying download folder to $path/loyaltyCardsOpen"
-cp -ru ./* "$path/loyaltyCardsOpen"
+
+# remove any existing backup files and ignore any errors
+rm "$path/loyaltyCardsOpen/*.bak" 2> /dev/null
```

Comment:
When is a bak file created and why it's interesting to remove it?

Reply:
Create backup files when making a copy of the new installation files:
`# copy the updated files`
`cp -ruS .bak ./* "$path/loyaltyCardsOpen"`

then copy back the database file if required, thus preserving the original database
`mv "$path/loyaltyCardsOpen/loyaltyCardsDb.db.bak" "$path/loyaltyCardsOpen/loyaltyCardsDb.db"`
the .new files make sure that a database still exists in case of any errors.
Any residual .bak files are removed before starting a new install/update.